superset/assets/src/dashboard/reducers/getInitialState.js

```diff
@@ -92,6 +86,26 @@ export default function(bootstrapData) {
       };
 
       sliceIds.add(key);
+
+      // if chart is newly added from explore view,
+      // add a row in layout
+      if (!chartIdToLayoutId[key] && layout[parentId]) {
+        const parent = layout[parentId];
+        const rowContainer = getRowContainer();
+        layout[rowContainer.id] = rowContainer;
+        parent.children.push(rowContainer.id);
+
+        const chartHolder = getChartHolder({
```

Comment:
I think it'd be better to use the `newComponentFactory` util function here to keep things consistent, and not need to hard code dimensions.

Reply:
yes. should use newComponentFactory.